src/redis-cli.c

```diff
@@ -3041,7 +3041,8 @@ static void usage(int err) {
 "  -X                 Read <tag> argument from STDIN (see example below).\n"
 "  -d <delimiter>     Delimiter between response bulks for raw formatting (default: \\n).\n"
 "  -D <delimiter>     Delimiter between responses for raw formatting (default: \\n).\n"
-"  -c                 Enable cluster mode (follow -ASK and -MOVED redirections).\n"
+"  -c                 Enable cluster mode (follow -ASK and -MOVED redirections\n"
+"                     and make --bigkeys and --memkeys usable on replicas).\n"
```

Comment:
i'm not certain this should be documented.
i.e. if the user connects to a replica, then that's what they intended.
or if we document that for the benefit of users that are trying to use --bigkeys, and don't know how to make it work?
in that case, maybe we better always send READONLY and ignore the error (rather than require `-c` from the user)?

Reply:
@oranagra Should we send READONLY even to non-cluster nodes?

Reply:
this command doesn't work on the entire cluster, right? it just scans the one node which you connected to, and unlike other things with cluster mode, it doesn't handle redirection, topology discovery, etc.
so i think we don't have to require the user to ask for cluster mode.
i don't see any disadvantage in sending READONLY even in non-cluster mode and just ignore the reply.
am i missing anything?

Reply:
You're right. Only think I can think of is if READONLY gives an error in non-cluster mode, it adds to error statistics, but that's no big deal.

Reply:
I don't know if what I said is right, but In fact,I think if -c is not included, there are about three ways. The first is to use rediscommand's info cluster to determine whether it is cluster mode. Maybe actually -c is unnecessary. All -c seems to be able to be determined through info cluster. The second is to ignore the corresponding readonly error in non-cluster mode in redis-cli.c. The third is to ignore the command in cluster.c to report the error in non-cluster mode.
As for "-c", when users use redis-cli, generally when connecting to cluster mode, users will add "-c" regardless of whether they need to add it.

Reply:
if ignore the error in redis-cli.c,maybe like this 
(read_reply->type == REDIS_REPLY_ERROR && strcmp(read_reply->str,"ERR This instance has cluster support disabled")) {
            fprintf(stderr, "Error: %s\n", read_reply->str);
            exit(1);
about those three ways，I'm not sure which one is the style of redis code.Please give me some advice， @oranagra

Reply:
If we want to ignore the error, we can't do `exit(1)`. :face_with_spiral_eyes: But we can do nothing, not even print the error. I think we can just remove this `else if`.

Reply:
I change the code @zuiderkwast ,please check.I recompile and execute it locally and it can pass basic functional tests.